CodeStarterCamp_Week5/main.swift

```diff
@@ -1,12 +1,115 @@
 //
 //  main.swift
-//  CodeStarterCamp_Week5
+//  Week5 Step1
 //
-//  Created by yagom.
-//  Copyright © yagom academy. All rights reserved.
+//  Created by JeonSangHyeok on 2023/03/21.
 //
 
 import Foundation
 
-print("Hello, World!")
+class Person {
+    var name: String
+    var height: Int
+    
+    init(name: String, height: Int) {
+        self.name = name
+        self.height = height
+    }
+}
 
+enum Level {
+    case A
+    case B
+    case C
+}
+
+protocol Talent {
+    var singing: Level { get set}
+    var dancing: Level { get set}
+    var acting: Level { get set}
+}
+
+protocol BadPersonality {
+    var frequancyOfCursing: Level { get set}
+}
+
+
+class TalentedPerson: Person, Talent {
+    var singing: Level
+    var dancing: Level
+    var acting: Level
+    
+    init(name: String, height: Int, singing: Level, dancing: Level, acting: Level) {
+        self.singing = singing
+        self.dancing = dancing
+        self.acting = acting
+        
+        super.init(name: name, height: height)
+    }
+}
+
+class TalentedPersonWithBadPersonality: Person, Talent, BadPersonality {
+    var singing: Level
+    var dancing: Level
+    var acting: Level
+    var frequancyOfCursing: Level
+    
+    init(name: String, height: Int, singing: Level, dancing: Level, acting: Level, frequancyOfCursing: Level) {
+        self.singing = singing
+        self.dancing = dancing
+        self.acting = acting
+        self.frequancyOfCursing = frequancyOfCursing
+        
+        super.init(name: name, height: height)
+    }
+}
+
+struct AuditionManager {
+    var totalApplicantsList: [Person]
+    var passedApplicantsList: [Person] = []
+    
+    func checkTalent(_ applicants: Talent) -> Bool {
+        return applicants.singing == .A || applicants.acting == .A || applicants.dancing == .A
+    }
+    
+    func checkPersonailty(_ appicants: Talent) -> Bool {
+        return appicants is BadPersonality
+    }
+    
+    mutating func cast() {
+        for applicants in totalApplicantsList {
+            guard let checkApplicants = applicants as? Talent else {
+                continue
+            }
+            if checkTalent(checkApplicants) {
+                guard checkPersonailty(checkApplicants) else {
+                    passedApplicantsList.append(applicants)
+                    continue
+                }
+            }
+        }
+    }
```

Comment:
```suggestion
    mutating func cast() {
        for applicants in totalApplicantsList {
            guard let checkApplicants = applicants as? Talent else {
                continue
            }
            
            if checkTalent(checkApplicants) {
                guard checkPersonailty(checkApplicants) else {
                    passedApplicantsList.append(applicants)
                    continue
                }
            }
        }
    }
```
컨벤션 차이기는 하지만 저는 가독성을 위해 개행을 하는 편입니다!

Reply:
좋은 정보 감사합니다. 다음부터 적용해봐야 겠네요.